pyccl/neutrinos.py

```diff
@@ -56,6 +52,12 @@ def Omega_nu_h2(a, *, m_nu, T_CMB=None, T_ncdm=None):
     return OmNuh2
 
 
+def OmNuh2(cosmo, a):
+    """Like Omega_nu_h2 but it uses the parameters from a Cosmology object."""
+    return Omega_nu_h2(a, m_nu=cosmo["m_nu"],
+                       T_CMB=cosmo["T_CMB"], T_ncdm=cosmo["T_ncdm"])
+
+
```

Comment:
So now we have `Omega_nu_h2`, `Omeganuh2` and `OmNuh2`. Any way we can clean this up?

Reply:
Three proposals:
- `omega_nu` (as in little omega)
- `little_Omega_nu`
- Just a private method of ccl. Users should instead use the `omega_x` and `rho_x` methods.

Reply:
My actual final proposal: we make `Omega_nu_h2` a private method of `Cosmology`, only used for initialization, and deprecate all other versions of this function in favour of `omega_x`.

Reply:
What about `nu_masses`?

Reply:
Let's leave `nu_masses` there, and just deprecate `T_CMB` as an argument.

Reply:
After your input and some modifications by me, there's now a private method in `Cosmology` which calculates OmNuh2, with the original implementation deprecated in favor of `omega_x`.